### PR TITLE
[jvm-packages] delete all constraints from spark layer about obj and eval metrics and handle error in jvm layer

### PR DESCRIPTION
**Pull request by CodingCat**
this PR deletes all constraints from spark layer about obj and eval

and simultaneously it ensures that the error caused by native layer (such as unsupported obj and eval) can be handled properly in jvm layer

### CODE REVIEW

**Comment by trivialfis** [comment]
Does it have anything to do with: https://github.com/dmlc/xgboost/blob/7f0f96e2f208dee8abb8f0a552268880a3702a63/jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/params/LearningTaskParams.scala#L107 ?

**Comment by CodingCat** [comment]
yes, in future, we don’t have such headaches anymore, whatever supported in c++ would be supported in spark

**Comment by trivialfis** [comment]
That would be great!

**Comment by CodingCat** [comment]
during the development, I found more and more problems in different layers of xgboost/rabit regarding fault handling, I will separate this into multiple PRs addressing them one by one

**Comment by CodingCat** [comment]
@sperlingxx can you help to review this?

**Comment by sperlingxx** [comment]
@CodingCat I'm grad to, and I'll review this tomorrow.

File: jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala [hunk 1]
```diff
@@ -292,7 +292,9 @@ class XGBoostClassificationModel private[ml](
 
         private val batchIterImpl = rowIterator.grouped($(inferBatchSize)).flatMap { batchRow =>
           if (batchCnt == 0) {
-            val rabitEnv = Array("DMLC_TASK_ID" -> TaskContext.getPartitionId().toString).toMap
+            val rabitEnv = Array(
+              "DMLC_TASK_ID" -> TaskContext.getPartitionId().toString,
+              "DMLC_WORKER_STOP_PROCESS_ON_ERROR" -> "false").toMap
```
Comment (sperlingxx) [inline jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala]: `DMLC_WORKER_STOP_PROCESS_ON_ERROR`,  what's the meaning of this tag?

**Comment by sperlingxx** [comment]
LGTM

File: jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala (hunk 1 above)
Comment (CodingCat) [reply jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala]: it's newly added in https://github.com/dmlc/rabit/blob/65b718a5e786bd7d0a850f3fa1df0dbdab023eb1/src/allreduce_base.cc

**Comment by CodingCat** [comment]
thanks for the review!

**Pull request merged**